### PR TITLE
Feature/for command

### PR DESCRIPTION
**Pull request by SgtSilvio**
- Added ForCommand
- Added ForCommandDe/Encoder
- Added DisconnectCommandDe/Encoder
- Refactoring

TODO: Tests for ForCommandDe/Encoder need are missing, this PR now for a faster review

### CODE REVIEW

File: src/main/java/org/varroa/chunk/command/Command.java [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+        DISCONNECT,
+        PUBLISH,
+        RUN,
+        SUBSCRIBE,
+        UNSUBSCRIBE,
+        WAIT,
+        FOR
+    }
+
+    public abstract @NotNull CommandType getType();
```
Comment (s-gheldd) [inline src/main/java/org/varroa/chunk/command/Command.java]: Nice!

File: src/main/java/org/varroa/chunk/command/Command.java [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
 
+    public enum CommandType {
+        CONNECT,
+        DISCONNECT,
+        PUBLISH,
+        RUN,
+        SUBSCRIBE,
+        UNSUBSCRIBE,
+        WAIT,
+        FOR
```
Comment (s-gheldd) [inline src/main/java/org/varroa/chunk/command/Command.java]: Nitpick: Alphabetical ordering would be FOR between DISCONNECT and PUBLISH

File: src/test/java/org/varroa/remote/codec/TestDeEncoders.java [hunk 3]
```diff
@@ -11,24 +11,30 @@
 public class TestDeEncoders {
 
     public static @NotNull CommandDecoders getDecoders() {
-        return new CommandDecoders(
+        final CommandDecoders[] commandDecoders = new CommandDecoders[1];
+        commandDecoders[0] = new CommandDecoders(
```
Comment (s-gheldd) [inline src/test/java/org/varroa/remote/codec/TestDeEncoders.java]: This is ugly as hell. Why not use a field?

File: src/test/java/org/varroa/remote/codec/TestDeEncoders.java [hunk 4]
```diff
@@ -11,24 +11,30 @@
...
+                new WaitCommandDecoder(),
+                new ForCommandDecoder(() -> commandDecoders[0]),
+                new RunCommandDecoder());
+        return commandDecoders[0];
     }
 
     public static @NotNull CommandEncoders getEncoders() {
-        return new CommandEncoders(
+        final CommandEncoders[] commandEncoders = new CommandEncoders[1];
+        commandEncoders[0] = new CommandEncoders(
```
Comment (s-gheldd) [inline src/test/java/org/varroa/remote/codec/TestDeEncoders.java]: See comment above.

File: src/test/java/org/varroa/remote/codec/TestDeEncoders.java (hunk 3 above)
Comment (SgtSilvio) [reply src/test/java/org/varroa/remote/codec/TestDeEncoders.java]: A field is not working as we can not pass an uninitialized field to the constructor.
The proper way would be to create a Provider the same way a dependency injection framework would do

Comment (s-gheldd) [reply src/test/java/org/varroa/remote/codec/TestDeEncoders.java]: Ah I did not know that.

**Pull request merged**